service/history/execution/mutable_state_builder.go

```diff
@@ -1772,6 +1772,11 @@ func (e *mutableStateBuilder) eventsToReplicationTask(
 
 	// the visibility timestamp will be set in shard context
 	replicationTask := &persistence.HistoryReplicationTask{
+		WorkflowIdentifier: persistence.WorkflowIdentifier{
```

Comment:
Consider adding a validation layer in relevant methods in persistence layer to return err when `WorkflowIdentifier` is not fully populated. (or turn that field into pointer to do `!= nil` check)

Reply:
Suggestion / out of scope of this PR/hotfix, but if this function was instead a method on history events, we could do better round-trip testing

ie, 
```
func(e HistoryEvent) ToReplicationTask() persitence.Task {
    ... 
}
```

```
func(e persitenceTask) ToHistoryEvent() types.HistoryEvent {
    ... 
}
```

(And onwards to roundtrip-testability)